source/dub/dependency.d

```diff
@@ -327,9 +326,6 @@ struct Dependency {
 		Dependency dep;
 		if( verspec.type == Json.Type.object ){
 			if( auto pp = "path" in verspec ) {
-				if (auto pv = "version" in verspec)
-					logDiagnostic("Ignoring version specification (%s) for path based dependency %s", pv.get!string, pp.get!string);
```

Comment:
move this log message to where it should be logged instead please, I think this one has value since it indicates that the recipe / selections isn't properly defined.

Reply:
There is nowhere to move it without exposing the content of the JSON object. Note that if we want to be strict about the content of the selection files (which I want to be!) it's better to have a generic mean of doing that (hence my previous work on the parser) rather than ad-hoc here and there.

Also note that this message is unlikely to be helpful because it's a `logDiagnostic` and hence only shows with `-v`.